src/motion_database.rs

```diff
@@ -0,0 +1,101 @@
+use bevy::prelude::*;
+use bvh_anim::{Frame, Joint};
+use serde::{Deserialize, Serialize};
+use std::{fs, io::Write};
+
+use crate::{
+    bvh::{bvh_asset::BvhAsset, bvh_player::get_pose},
+    ui::BuildConfig,
+};
+
+pub struct MotionDatabasePlugin;
+
+impl Plugin for MotionDatabasePlugin {
+    fn build(&self, _app: &mut App) {}
+}
+
+pub type Pose = Vec<Vec<f32>>;
+
+#[derive(Serialize, Deserialize)]
+pub struct TrajectoryPosition {
+    pub position: Vec3,
+    pub time: f32,
+}
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct MotionData {
+    pub trajectories: Vec<TrajectoryPosition>,
+    pub trajectory_offsets: Vec<usize>,
+    pub poses: Vec<Pose>,
+    pub pose_offset: Vec<usize>,
+}
+
+// trajectories: [ bvh0:traj0, bvh0:traj1, bvh0:traj2, bvh1:traj0, bvh1:traj1, bvh2:traj0 ]
+//      offsets: [ 0, 3, 5, 6 ]
+
+pub fn extract_motion_data(bvh_asset: &Assets<BvhAsset>, build_config: &mut BuildConfig) {
+    let mut motion_data = MotionData::default();
+    for id in build_config.bvh_assets.iter() {
+        println!("id: {}", id);
+        let Some(BvhAsset(bvh)) = bvh_asset.get(*id) else {
+            return;
+        };
+
+        let interval = 0.3333;
```

Comment:
To check if code works correctly, you can try to increase this interval size to some large number like 50.0. Select 2~3 files and press build, and check the json there. Maybe can post one in the comment here also :D.

Reply:
![image](https://github.com/kahboon0425/bevy_motion_matching/assets/116237965/33793929-0c2d-43fe-b65e-0fca53b68ca2) the output seems to be correct

Reply:
ok, nice, only thing left now is to do a sum scan operation in each iteration, I will make another review pointing it out.